File: backend/src/Linuxdle.Domain/DailyCommands/CommandInfo.cs
namespace Linuxdle.Domain.DailyCommands;

public sealed class CommandInfo
{
    private CommandInfo() { }

    public int Id { get; private set; }
    public int CommandId { get; private set; }
    public string Description { get; private set; } = string.Empty;
    public string Synopsis { get; private set; } = string.Empty;
    public string Example { get; private set; } = string.Empty;
    public string FunFact { get; private set; } = string.Empty;

    public DailyCommand Command { get; private set; } = null!;

    public static CommandInfo Create(
        int commandId,
        string description,
        string synopsis,
        string example,
        string funFact)
    {
        return new()
        {
            CommandId = commandId,
            Description = description,
            Synopsis = synopsis,
            Example = example,
            FunFact = funFact
        };
    }
}

File: seed_data/command_info.py
"""Seed data for the command_info table."""

import json
import os
from psycopg2.extras import execute_values


def get_command_info_data():
    """Return command info seed data from JSON file."""
    data_path = os.path.join(os.path.dirname(__file__), "data", "command_info.json")
    with open(data_path, "r") as f:
        infos = json.load(f)
    return [
        (i["command_id"], i["description"], i["synopsis"], i["example"], i["fun_fact"])
        for i in infos
    ]


def seed_command_info(cur):
    """Seed the command_info table."""
    print("Seeding 'command_info'...")
    data = get_command_info_data()
    execute_values(
        cur,
        """
        INSERT INTO command_infos (command_id, description, synopsis, example, fun_fact)
        VALUES %s
        ON CONFLICT (command_id) DO UPDATE SET
            description = EXCLUDED.description,
            synopsis = EXCLUDED.synopsis,
            example = EXCLUDED.example,
            fun_fact = EXCLUDED.fun_fact
        """,
        data,
    )
